### PR TITLE
Provide suggestions for embedded Sass styles

### LINKED ISSUE #98: Need suggestions for embedded Sass styles
Current version of `autocomplete-css` [detect whether the scope is a Sass one](https://github.com/atom/autocomplete-css/blob/master/lib/provider.coffee#L29) by using the method [hasScope](https://github.com/atom/autocomplete-css/blob/master/lib/provider.coffee#L302-L303) to search the scope array to find one with the exact name `source.sass`. Such behaviour makes autocompletion broken in many embedding situations.

Take `.vue` as an example (hedefalk/atom-vue#46), lines between `style` tags have no suggestions for Sass:

```vue
<template>
  <!-- The scope here is [text.html.vue] -->
</template>

<style lang="sass">
  // The scopes here are [text.html.vue, source.sass.embedded.html]
html
  widt// no suggestion
</style>
```

We could hack this issue by using `source.sass` as the scope name in `text.html.vue`. But doing that will violate the scope naming conventions which we respect a lot.

I think we could improve the way of scope detection here by change `hasScope` to `containScope` which search the scope array to find any scope starts with `source.sass`. Any idea?

### PR DESCRIPTION
**Pull request by aidistan**
### Description of the Change

This PR aims to fix #98.

### Alternate Designs

Please refer to the discussion in #98.

### Benefits

Recognize embedded Sass scopes and provide suggestions for them.

### CODE REVIEW

File: lib/provider.coffee [hunk 1]
```diff
@@ -299,8 +299,9 @@ lineEndsWithSemicolon = (bufferPosition, editor) ->
   line = editor.lineTextForBufferRow(row)
   /;\s*$/.test(line)
 
-hasScope = (scopesArray, scope) ->
-  scopesArray.indexOf(scope) isnt -1
+hasScope = (scopesArray, scope, checkEmbedded = false) ->
+  scopesArray.indexOf(scope) isnt -1 or
+    (checkEmbedded and scopesArray.indexOf(scope + '.embedded.html') isnt -1)
```
Comment (winstliu) [inline lib/provider.coffee]: :art: `(checkEmbedded and scopesArray.indexOf("#{scope}.embedded.html") isnt -1)`

**Comment by winstliu** [comment]
Should all top-level scopes check embedded?  i.e. `source.css`, `source.css.less`, `source.css.scss`.

**Comment by aidistan** [comment]
@50Wliu You are right. I have fixed them too.

**Comment by winstliu** [comment]
Thanks!  Normally I would require tests for a change like this, but I see how it would be hard to test without bringing in the Vue package as a test dependency.

**Pull request merged**